src/utils/address-utils.ts

```diff
@@ -0,0 +1,1176 @@
+          parameters: []
+        }
+      }
+    default:
+      throw Error(`Mapping not supported for ${mappingId}`)
+  }
+}
+
+// You should never need to edit this function.  If there is a bug here raise an issue in [Github](https://github.com/opencrvs/opencrvs-farajaland)
+export function getIdentifiersFromFieldId(fieldId: string) {
+  const splitIds = fieldId.split('.')
+  return {
+    event: splitIds[0] as Event,
+    sectionId: splitIds[1],
+    groupId: splitIds[2],
+    fieldName: splitIds[3]
+  }
+}
+
+// You should never need to edit this function.  If there is a bug here raise an issue in [Github](https://github.com/opencrvs/opencrvs-farajaland)
+export function getSectionIdentifiers(fieldId: string, form: ISerializedForm) {
+  const { event, sectionId } = getIdentifiersFromFieldId(fieldId)
+
+  const sectionIndex = form.sections.findIndex(({ id }) => id === sectionId)
+  return {
+    event,
+    sectionIndex,
+    sectionId
+  }
+}
+
+// You should never need to edit this function.  If there is a bug here raise an issue in [Github](https://github.com/opencrvs/opencrvs-farajaland)
+export function getGroupIdentifiers(fieldId: string, form: ISerializedForm) {
+  const { event, groupId } = getIdentifiersFromFieldId(fieldId)
+
+  const { sectionIndex, sectionId } = getSectionIdentifiers(fieldId, form)
+
+  const groups = form.sections[sectionIndex].groups
+
+  const groupIndex = groups.findIndex(({ id }) => id === groupId)
+
+  return {
+    event,
+    sectionIndex,
+    sectionId,
+    groupIndex
+  }
+}
+
+// You should never need to edit this function.  If there is a bug here raise an issue in [Github](https://github.com/opencrvs/opencrvs-farajaland)
+export function getFieldIdentifiers(fieldId: string, form: ISerializedForm) {
+  const { event, fieldName } = getIdentifiersFromFieldId(fieldId)
+
+  const { sectionIndex, groupIndex, sectionId } = getGroupIdentifiers(
+    fieldId,
+    form
+  )
+
+  const fields = form.sections[sectionIndex].groups[groupIndex].fields
+
+  const fieldIndex = fields.findIndex(({ name }) => name === fieldName)
+
+  return {
+    event,
+    sectionIndex,
+    sectionId,
+    groupIndex,
+    fieldIndex
+  }
+}
+
+// You should never need to edit this function.  If there is a bug here raise an issue in [Github](https://github.com/opencrvs/opencrvs-farajaland)
+export function isUseCaseForPlaceOfEvent(useCase: string): Boolean {
+  return Object.values(
+    EventLocationAddressCases as Record<string, string>
+  ).includes(useCase)
+}
+
+// You should never need to edit this function.  If there is a bug here raise an issue in [Github](https://github.com/opencrvs/opencrvs-farajaland)
+export const getAddressSubsection = (
+  previewGroup: AddressSubsections,
+  label: MessageDescriptor,
+  conditionalCase?: string
+): SerializedFormField[] => {
+  const fields: SerializedFormField[] = []
+  const subsection: SerializedFormField = {
+    name: previewGroup,
+    type: 'SUBSECTION_HEADER',
```

Comment:
This should be 'HEADING3'. https://github.com/opencrvs/opencrvs-farajaland/blob/b62acc16ae979a58dabd01fa0acd8a948a70122b/src/form/addresses/address-utils.ts#L503-L528